app/actions/local/channel.ts

```diff
@@ -237,32 +239,51 @@ export async function resetMessageCount(serverUrl: string, channelId: string) {
     }
 }
 
-export async function storeMyChannelsForTeam(serverUrl: string, teamId: string, channels: Channel[], memberships: ChannelMembership[], prepareRecordsOnly = false, isCRTEnabled?: boolean) {
+const resolveAndflattenModels = async (operator: ServerDataOperator, modelPromises: Array<Promise<Model[]>>, description: string, prepareRecordsOnly: boolean) => {
+    const models = await Promise.all(modelPromises);
+    if (!models.length) {
+        return {models: []};
+    }
+
+    const flattenedModels = models.flat();
+
+    if (prepareRecordsOnly) {
+        return {models: flattenedModels};
+    }
+
+    if (flattenedModels.length) {
+        await operator.batchRecords(flattenedModels, description);
+    }
+
+    return {models: flattenedModels, error: undefined};
+};
+
+export async function storeAllMyChannels(serverUrl: string, channels: Channel[], memberships: ChannelMembership[], isCRTEnabled: boolean, prepareRecordsOnly = false) {
     try {
         const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const cs = new Set(channels.map((c) => c.id));
         const modelPromises: Array<Promise<Model[]>> = [
-            ...await prepareMyChannelsForTeam(operator, teamId, channels, memberships, isCRTEnabled),
+            ...await prepareAllMyChannels(operator, channels, memberships.filter((m) => cs.has(m.channel_id)), isCRTEnabled),
         ];
 
-        const models = await Promise.all(modelPromises);
-        if (!models.length) {
-            return {models: []};
-        }
-
-        const flattenedModels = models.flat();
-
-        if (prepareRecordsOnly) {
-            return {models: flattenedModels};
-        }
+        return resolveAndflattenModels(operator, modelPromises, 'storeAllMyChannels', prepareRecordsOnly);
+    } catch (error) {
+        logError('Failed storeAllMyChannels', error);
+        return {error, models: undefined};
+    }
+}
 
-        if (flattenedModels.length) {
-            await operator.batchRecords(flattenedModels, 'storeMyChannelsForTeam');
-        }
+export async function storeMyChannelsForTeam(serverUrl: string, teamId: string, channels: Channel[], memberships: ChannelMembership[], prepareRecordsOnly = false, isCRTEnabled?: boolean) {
+    try {
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const modelPromises: Array<Promise<Model[]>> = [
+            ...await prepareMyChannelsForTeam(operator, teamId, channels, memberships, isCRTEnabled),
+        ];
 
-        return {models: flattenedModels};
+        return resolveAndflattenModels(operator, modelPromises, 'storeMyChannelsForTeam', prepareRecordsOnly);
```

Comment:
Same comment about returning promises
```suggestion
        return await resolveAndflattenModels(operator, modelPromises, 'storeMyChannelsForTeam', prepareRecordsOnly);
```